### PR TITLE
Update user guide with new features and extensions

### PR DESCRIPTION
**Pull request by zhuuyicheng**
- Added new feature `filter`
- Added images to the new feature `filter`, including for the extensions like `help` and `clear`
- Added navigation guide at top of page

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+layout: default.md
+title: "User Guide"
+pageNav: 3
 ---
 
-# AB-3 User Guide
+# CulinaryContacts User Guide
 
-AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized for use via a  Line Interface** (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB3 can get your contact management tasks done faster than traditional GUI apps.
+CulinaryContacts is a **desktop app for managing contacts, optimized for use via a  Line Interface** (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB3 can get your contact management tasks done faster than traditional GUI apps.
```
Comment (KSSWSept20) [inline docs/UserGuide.md:9]: I think it is better to change AB3 to CulinaryContacts here.

**Review by KSSWSept20 — CHANGES_REQUESTED** [review]
Overall, LGTM, except for the first part of the general description of the application. I think changing AB3 to CulinaryContacts will be better.

File: docs/UserGuide.md (hunk 1 above)
Comment (zhuuyicheng) [reply docs/UserGuide.md:9]: Nice catch, I missed it my bad

File: docs/UserGuide.md [hunk 2]
```diff
@@ -128,9 +142,27 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
+### Filtering persons by tag: `filter`
+
+Finds persons that are tagged with any of the given tags.
+
+Format: `filter TAG [MORE_TAGS]`
+
+* The search is case-insensitive. e.g `friends` will match with `Friends`.
+* The order of the tags does not matter. e.g. `friends family` will match `family friends`.
+* Only the tag is searched.
+* Only full tags will be matched e.g. `friend` will not match `friends`
```
Comment (ziiqii) [inline docs/UserGuide.md:157]: Maybe add period on line 157 for consistency.

**Review by ziiqii — COMMENTED** [review]
LGTM. Just a few nits to fix.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+    * [Viewing help:`help`](#viewing-help--help)
+    * [Adding a person: `add`](#adding-a-person-add)
+    * [Listing all persons: `list`](#listing-all-persons--list)
+    * [Editing a person: `edit`](#editing-a-person--edit)
+    * [Locating persons by name: `find`](#locating-persons-by-name-find)
+    * [Filtering persons by tag: `filter`](#filtering-persons-by-tag-filter)
+    * [Deleting a person: `delete`](#deleting-a-person--delete)
+    * [Exiting the program: `exit`](#exiting-the-program--exit)
+    * [Saving the data](#saving-the-data)
+    * [Editing the data file](#saving-the-data)
```
[the remark this answers is not in the document]
Comment (zhuuyicheng) [reply docs/UserGuide.md:23]: Yes, you are correct.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
 <!-- * Table of Contents -->
-<page-nav-print />
-
+* [Quickstart](#quick-start)
+* [Features](#features)
+    * [Viewing help:`help`](#viewing-help--help)
+    * [Adding a person: `add`](#adding-a-person-add)
+    * [Listing all persons: `list`](#listing-all-persons--list)
+    * [Editing a person: `edit`](#editing-a-person--edit)
+    * [Locating persons by name: `find`](#locating-persons-by-name-find)
```
[the remark this answers is not in the document]
Comment (zhuuyicheng) [reply docs/UserGuide.md:18]: Done, changed all instances of 'locate' to 'find'.

File: docs/UserGuide.md (hunk 2 above)
Comment (zhuuyicheng) [reply docs/UserGuide.md:157]: Added period on all sentences.

**Review by ziiqii — APPROVED** [review]
LGTM!

**Review by KSSWSept20 — APPROVED** [review]
LGTM overall

**Pull request merged**